File: sdk/android/src/main/java/com/dualler/sdk/handler/DeviceHandler.kt
package com.dualler.sdk.handler

import android.content.Context
import android.os.Build
import com.dualler.sdk.protocol.APIContract
import com.dualler.sdk.protocol.BridgeMessage

class DeviceHandler(private val context: Context) : Handler {
    override val apiName: String = APIContract.GET_SYSTEM_INFO

    override fun handle(message: BridgeMessage): String? {
        return """{
            "brand":"${Build.BRAND}",
            "model":"${Build.MODEL}",
            "system":"Android ${Build.VERSION.RELEASE}",
            "platform":"android",
            "sdkVersion":"${Build.VERSION.SDK_INT}"
        }""".replace("\n", "").replace(", ", "")
    }
}

File: sdk/ios/Sources/DuallerSDK/DeviceHandler.swift
import UIKit

/// Returns device and screen information
public final class DeviceHandler: Handler {
    public let apiName: String = "dualler.getSystemInfo"

    public func handle(_ message: BridgeMessage) -> String? {
        let screen = UIScreen.main.bounds
        let device = UIDevice.current
        let statusBarHeight = UIApplication.shared.statusBarFrame.height

        return String(
            format: "{\"brand\":\"%@\",\"model\":\"%@\",\"system\":\"%s %@\",\"platform\":\"ios\",\"screenWidth\":%.0f,\"screenHeight\":%.0f,\"windowWidth\":%.0f,\"windowHeight\":%.0f,\"statusBarHeight\":%.0f,\"pixelRatio\":%.1f}",
            device.model as CFString,
            device.name,
            "iOS",
            device.systemVersion,
            screen.width,
            screen.height,
            screen.width,
            screen.height,
            Double(statusBarHeight),
            UIScreen.main.scale
        )
    }
}
